Domoticz still crashes on exit with Python 3.9.1 (was 3.9.0)

```
2020-11-24 11:14:22.893  Status: Domoticz V2020.2 (build 12701) (c)2012-2020 GizMoCuz
2020-11-24 11:14:22.893  Status: Build Hash: 152a10583-modified, Date: 2020-11-24 07:54:29
2020-11-24 11:14:22.893  Status: Startup Path: /home/admin/dev-domoticz/
2020-11-24 11:14:23.127  Status: PluginSystem: Started, Python version '3.9.0'.
2020-11-24 11:14:23.128  Active notification Subsystems: http (1/13)
2020-11-24 11:14:23.136  Status: WebServer(HTTP) started on address: :: with port 8080
2020-11-24 11:14:23.165  Status: WebServer(SSL) started on address: :: with port 443
2020-11-24 11:14:23.166  Starting shared server on: :::6144
2020-11-24 11:14:23.166  Status: TCPServer: shared server started...
2020-11-24 11:14:23.166  Status: RxQueue: queue worker started...
2020-11-24 11:14:25.194  Status: evohome: Controller ID (0x53376)
2020-11-24 11:14:25.194  Status: NotificationSystem: thread started...
2020-11-24 11:14:25.194  Status: EventSystem: reset all events...
2020-11-24 11:14:25.249  Status: EventSystem: reset all device statuses...
2020-11-24 11:14:25.279  Status: Python EventSystem: Initalizing event module.
2020-11-24 11:14:25.279  Status: EventSystem: Started
2020-11-24 11:14:25.279  Status: EventSystem: Queue thread started...
2020-11-24 11:14:25.617  Status: PluginSystem: Entering work loop.
2020-11-24 11:14:26.194  Status: evohome serial: Opening serial port: /dev/ttyUSB-HGI80@115200

```

On pressing CTRL+C on the cmd line I am greeted with


```
2020-11-24 11:15:04.670  Status: WebServer(HTTP) stopped
2020-11-24 11:15:05.170  Status: WebServer(SSL) stopped
2020-11-24 11:15:05.171  Status: TCPServer: shared server stopped
2020-11-24 11:15:05.171  Status: Stopping all hardware...
2020-11-24 11:15:05.280  Status: Evohome: Worker stopped...
2020-11-24 11:15:05.280  Status: Scheduler stopped...
2020-11-24 11:15:05.280  Status: EventSystem: Stopped...
2020-11-24 11:15:05.280  Status: EventSystem: Queue thread stopped...
2020-11-24 11:15:05.281  Error: Domoticz(pid:26127, tid:26127('domoticz')) received fatal signal 11 (Segmentation fault)
2020-11-24 11:15:05.281  Error: siginfo address=0x10, address=0x7f67f12678b0
2020-11-24 11:15:07.109  Error: Thread 1 (Thread 0x7f67f83e4780 (LWP 26127)):
2020-11-24 11:15:07.109  Error: #0  0x00007f67f71d432a in __waitpid (pid=26148, stat_loc=0x7ffef9f78dfc, options=0) at ../sysdeps/unix/sysv/linux/waitpid.c:30
2020-11-24 11:15:07.109  Error: #1  0x00005570e86c8f7f in dumpstack_gdb(bool) ()
2020-11-24 11:15:07.109  Error: #2  0x00005570e86c9569 in signal_handler(int, siginfo_t*, void*) ()
2020-11-24 11:15:07.109  Error: #3  <signal handler called>
2020-11-24 11:15:07.109  Error: #4  0x00007f67f12678b0 in PyEval_ReleaseLock () from /usr/lib/x86_64-linux-gnu/libpython3.9.so
2020-11-24 11:15:07.109  Error: #5  0x00005570e864cf10 in Plugins::PythonEventsStop() ()
2020-11-24 11:15:07.109  Error: #6  0x00005570e863970d in CEventSystem::StopEventSystem() ()
2020-11-24 11:15:07.109  Error: #7  0x00005570e8688fb6 in MainWorker::Stop() ()
2020-11-24 11:15:07.109  Error: #8  0x00005570e85c449f in main ()
Segmentation fault

```

I changed the PluginManager.h and it compiled OK on a system with Python 3.9.1
but link step results in:
```

/usr/bin/ld: CMakeFiles/domoticz.dir/hardware/plugins/PluginManager.cpp.o: warning: relocation against `_ZN7Plugins13CPluginSystem21m_InitialPythonThreadE' in read-only section `.text'
/usr/bin/ld: CMakeFiles/domoticz.dir/main/EventsPythonModule.cpp.o: in function `Plugins::PythonEventsInitialize(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
EventsPythonModule.cpp:(.text+0xe05): undefined reference to `Plugins::CPluginSystem::m_InitialPythonThread'
/usr/bin/ld: CMakeFiles/domoticz.dir/hardware/plugins/Plugins.cpp.o: in function `Plugins::CPlugin::Initialise()':
Plugins.cpp:(.text+0x6fb8): undefined reference to `Plugins::CPluginSystem::m_InitialPythonThread'
/usr/bin/ld: CMakeFiles/domoticz.dir/hardware/plugins/Plugins.cpp.o: in function `Plugins::CPlugin::Stop()':
Plugins.cpp:(.text+0xbfe5): undefined reference to `Plugins::CPluginSystem::m_InitialPythonThread'
/usr/bin/ld: CMakeFiles/domoticz.dir/hardware/plugins/PluginManager.cpp.o: in function `Plugins::CPluginSystem::StopPluginSystem()':
PluginManager.cpp:(.text+0x2783): undefined reference to `Plugins::CPluginSystem::m_InitialPythonThread'
/usr/bin/ld: PluginManager.cpp:(.text+0x279b): undefined reference to `Plugins::CPluginSystem::m_InitialPythonThread'
/usr/bin/ld: CMakeFiles/domoticz.dir/hardware/plugins/PluginManager.cpp.o:PluginManager.cpp:(.text+0x2add): more undefined references to `Plugins::CPluginSystem::m_InitialPythonThread' follow
/usr/bin/ld: warning: creating DT_TEXTREL in a PIE
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/domoticz.dir/build.make:4317: bin/domoticz] Error 1
make[1]: *** [CMakeFiles/Makefile2:492: CMakeFiles/domoticz.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
```

--- a/main/EventsPythonModule.cpp
+++ b/main/EventsPythonModule.cpp
@@ -160,6 +160,7 @@
 			if (Plugins::Py_IsInitialized())
 				Py_EndInterpreter((PyThreadState *)m_PyInterpreter);
 			m_PyInterpreter = nullptr;
+			PyThreadState_Swap((PyThreadState *)m_mainworker.m_pluginsystem.PythonThread());
 			PyEval_ReleaseLock();
 			_log.Log(LOG_STATUS, "EventSystem - Python stopped...");
 			return true;

